src/main/kotlin/org/jqassistant/tooling/intellij/plugin/editor/report/ReportToolWindowContent.kt

```diff
@@ -0,0 +1,130 @@
+package org.jqassistant.tooling.intellij.plugin.editor.report
+
+import com.intellij.openapi.application.ApplicationManager.getApplication
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.fileEditor.FileEditorManager
+import com.intellij.openapi.fileEditor.OpenFileDescriptor
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.findFile
+import com.intellij.ui.components.JBScrollPane
+import com.intellij.ui.treeStructure.Tree
+import org.jqassistant.schema.report.v2.*
+import java.util.*
+import javax.swing.JPanel
+import javax.swing.event.TreeSelectionEvent
+
+
+open class ReportToolWindowContent(
+    private val project: Project,
+    private val baseDir: VirtualFile,
+    private val report: JqassistantReport
+) {
+    val contentPanel: JBScrollPane
+
+    init {
+        val projectTrees = treePanel()
+
+        val firstTree = projectTrees.first()
+        if (projectTrees.size == 1) {
+            contentPanel = JBScrollPane(firstTree)
+        } else {
+            val subPanel = JPanel()
+            for (tree in projectTrees) subPanel.add(tree)
+
+            contentPanel = JBScrollPane(subPanel)
+        }
+    }
+
+    private fun treePanel(): List<Tree> {
+        val nodeList = buildRuleTree(null, report.groupOrConceptOrConstraint)
+        val cellRenderer = ReportCellRenderer()
+
+        val treeList: ArrayList<Tree> = ArrayList()
```

Comment:
```suggestion
        val treeList: List<Tree> = listOf()
```
And use `nodeList.map` to prevent using a mutable list